plugin/pki/path_venafi_cert_enroll.go

```diff
@@ -438,6 +440,32 @@ func formRequest(reqData requestData, role *roleEntry, signCSR bool, logger hclo
 		return certReq, fmt.Errorf("Invalid chain option %s", role.ChainOption)
 	}
 
+	if role.TTL > 0 {
+		days := float64(role.TTL.Hours()) / 24
+		roundedDays := math.Round(float64(days)) //round days to convert them on the nearest days, based on the hours
+
+		issuerHint := ""
+
+		if role.IssuerHint != "" {
+			issrOpt := string(role.IssuerHint[0])
+			issrOpt = strings.ToLower(issrOpt)
+
+			switch issrOpt {
+			case "m":
+				issuerHint = util.IssuerHintMicrosoft
+			case "d":
+				issuerHint = util.IssuerHintDigicert
+			case "e":
+				issuerHint = util.IssuerHintEntrust
+
+			}
+
+		}
+
+		certReq.IssuerHint = issuerHint
+		certReq.ValidityHours = int(roundedDays) * 24
```

Comment:
This seems a bit odd, do we need to round to the nearest day and then convert back to hours?  I was expecting that VCert would handle that since the nearest day rounding is meant to only apply to TPP.  The fact that `ttl` in Vault is expressed in hours was actually the basis for the decision to have the validity expressed in hours in VCert (i.e. "ValidityHours").  Ultimately the target CA decides if/how it rounds the requested date.